File: Add Code Here/LeetCode_Sols/Medium/1996. The Number of Weak Characters in the Game/Solution.py
class Solution(object):
    def numberOfWeakCharacters(self, properties):
        count = 0
        maxN = ~sys.maxint
        properties.sort(key=lambda x: (-x[0], x[1]))
        for i in range(len(properties)):
            if(properties[i][1] < maxN):
                count+=1
            maxN = max(maxN, properties[i][1])
        return count

File: Add Code Here/LeetCode_Sols/Medium/1996. The Number of Weak Characters in the Game/Solution.cpp
class Solution {
public:
    static bool compare(vector<int>& a, vector<int>& b){
        if(a[0] == b[0])
            return a[1] < b[1];
        return a[0] > b[0];
    }
    
    int numberOfWeakCharacters(vector<vector<int>>& properties) {
        int n = properties.size();
        int count = 0;
        sort(properties.begin(), properties.end(), compare);
        int maxN = INT_MIN;
        for(int i = 0; i < n; i++){
            if(properties[i][1] < maxN)
                count++;
            maxN = max(maxN, properties[i][1]);
        }
        return count;
    }
};
